tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -397,11 +413,25 @@ def media_files(self):
 
     def delete_all_sms(self):
         self.marionette.switch_to_frame()
-        return self.marionette.execute_async_script("return GaiaDataLayer.deleteAllSms();", special_powers=True)
+        self.marionette.push_permission('sms', True)
```

Comment:
Good idea, I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1161606 for this.